integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,622 @@
+      Integration.orderCompleted = Integration.orderCompletedClassic;
+    } else if (Integration.options.gaOptions.enhancedEcommerce) {
+      Integration.productListViewed = Integration.productListViewedEnhanced;
+      Integration.productClicked = Integration.productClickedEnhanced;
+      Integration.productViewed = Integration.productViewedEnhanced;
+      Integration.productAdded = Integration.productAddedEnhanced;
+      Integration.productRemoved = Integration.productRemovedEnhanced;
+      Integration.promotionViewed = Integration.promotionViewedEnhanced;
+      Integration.promotionClicked = Integration.promotionClickedEnhanced;
+      Integration.checkoutStarted = Integration.checkoutStartedEnhanced;
+      Integration.orderRefunded = Integration.orderRefundedEnhanced;
+      Integration.orderCompleted = Integration.orderCompletedEnhanced;
+      Integration.checkoutStepViewed = Integration.checkoutStepViewedEnhanced;
+      Integration.checkoutStepCompleted =
+        Integration.checkoutStepCompletedEnhanced;
+      Integration.orderUpdated = Integration.orderUpdatedEnhanced;
+
+      // Integration.productListFiltered = Integration.productListFilteredEnhanced;
+    }
+    /* eslint-enable */
+});
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/gtagjs
+ *
+ * @api public
+ */
+
+GTAG.prototype.initialize = function() {
+  var config = [];
+  var that = this;
+  var gaOptions = this.options.gaOptions;
+  var GA_WEB_MEASUREMENT_ID = this.options.GA_WEB_MEASUREMENT_ID;
+  var GA_WEB_APP_MEASUREMENT_ID = this.options.GA_WEB_APP_MEASUREMENT_ID;
+  var AW_CONVERSION_ID = this.options.AW_CONVERSION_ID;
+  var DC_FLOODLIGHT_ID = this.options.DC_FLOODLIGHT_ID;
+  var accountId =
+    GA_WEB_MEASUREMENT_ID ||
+    GA_WEB_APP_MEASUREMENT_ID ||
+    AW_CONVERSION_ID ||
+    DC_FLOODLIGHT_ID;
+
+  if (GA_WEB_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (GA_WEB_APP_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_APP_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_APP_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (AW_CONVERSION_ID) {
+    config.push(['config', AW_CONVERSION_ID]);
+  }
+
+  if (DC_FLOODLIGHT_ID) {
+    config.push(['config', DC_FLOODLIGHT_ID]);
+  }
+
+  if (accountId) {
+    this.load({ accountId: accountId }, function() {
+      // Default routing.
+      for (var i = 0; i < config.length; i++) {
+        push(config[i][0], config[i][1]);
+      }
+      that.ready();
+    });
+  } else {
+    // Error case where not any of the ID specified
+  }
```

Comment:
More of a style nit - can we return early if `accountId` falsy to avoid nested code?
```suggestion
// Error case where not any of the ID specified
if (!accountId) {
    return;
}

this.load({ accountId: accountId }, function() {
    // Default routing.
    for (var i = 0; i < config.length; i++) {
      push(config[i][0], config[i][1]);
    }
    that.ready();
});
```

Reply:
Done